src/HID-APIs/TouchscreenAPI.h

```diff
@@ -0,0 +1,115 @@
+/*
+Copyright (c) 2021 ilufang
+See the readme for credit to other people.
+
+Permission is hereby granted, free of charge, to any person obtaining a copy
+of this software and associated documentation files (the "Software"), to deal
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in
+all copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+THE SOFTWARE.
+*/
+
+// Include guard
+#pragma once
+
+/// Maximum amount of fingers supported
+#define HID_TOUCHSCREEN_MAXFINGERS 10
+/// Number of fingers in a single report
+#define HID_TOUCHSCREEN_REPORTFINGERS 2
+
+// A report will always be the same size, even if you report fewer fingers than
+// REPORTFINGERS. The unused finger entries will simply be zero. If more fingers
+// are present than REPORTFINGERS, multiple reports will be sent to report all
+// fingers. This is know as "Hybrid Mode" on MSDN. The number of supported
+// fingers identified by Windows will still be MAXFINGERS. More than MAXFINGERS
+// contacts may be ignored by Windows even with hybrid mode.
+
+
+// Bit-mask flags for 'status' in the HID report
+#define HID_TOUCHSCREEN_TOUCH_CONTACT 0x01
+#define HID_TOUCHSCREEN_TOUCH_IN_RANGE 0x02
+
+typedef struct ATTRIBUTE_PACKED {
+	uint8_t status;
+	uint8_t pressure;
+	uint16_t x;
+	uint16_t y;
+} HID_Touchscreen_Finger_t;
+
+typedef union ATTRIBUTE_PACKED {
+	uint8_t whole8[0];
+	uint16_t whole16[0];
+	uint32_t whole32[0];
+	struct ATTRIBUTE_PACKED {
+		uint8_t count;
+		struct ATTRIBUTE_PACKED {
+			uint8_t identifier;
+			HID_Touchscreen_Finger_t touch;
+		} contacts[HID_TOUCHSCREEN_REPORTFINGERS];
+	};
+} HID_TouchscreenReport_Data_t;
+
+class TouchscreenAPI
+{
+public:
+
+	inline void begin();
+
```

Comment:
You have no end() function here, please add one. It should fully reset the touchscreen pressed.

Reply:
OK. Added.